Sources/Sentry/SentryTimeToDisplayTracker.m

```diff
@@ -128,13 +133,19 @@ - (void)framesTrackerHasNewFrame:(NSDate *)newFrameDate
         [self.initialDisplaySpan finish];
         if (!_waitForFullDisplay) {
             [SentryDependencyContainer.sharedInstance.framesTracker removeListener:self];
+#    if SENTRY_TARGET_PROFILING_SUPPORTED
+            stopLaunchProfile(nil);
```

Comment:
`h`: When TTFD is enabled, this stops the launch profile and the app start transaction when we report TTID. Instead, when TTFD is enabled the app start transaction must continue until the user calls TTFD. When the user doesn't call TTFD the tracer finishes after some timeout. It would be great to add tests for these use cases to `SentryTimeToDisplayTrackerTest`.

Reply:
I think calling ` stopLaunchProfile` in the `finishCallback` of the tracer here should do the job:

https://github.com/getsentry/sentry-cocoa/blob/69fcf6f477b69599124a017d5b819db58fa20629/Sources/Sentry/SentryTimeToDisplayTracker.m#L83-L84

Reply:
I don't understand this feedback, if TTFD is enabled, this branch won't be taken because of the check on line 134 for `if (!_waitForFullDisplay)`, right?

Reply:
> if TTFD is enabled, this branch won't be taken because of the check on line 134 for if (!_waitForFullDisplay), right?

Yes, that's true. Sorry, I had to finish my review yesterday quickly, cause I needed to watch my son. 

> When the user doesn't call TTFD the tracer finishes after some timeout.

This still applies. There is no guarantee that the user calls the manual TTFD API. When they don't call the manual TTFD API, the tracer times out and the `stopAndDiscardLaunchProfileTracer` after `[self.fullDisplaySpan finish];` will never be called. I think calling `stopAndDiscardLaunchProfileTracer` in the `finishCallback` should be the better approach.

Reply:
Is the TTFD timeout managed via `SentryTracer.deadlineTimerFired`? If so then that will stop the profiler as well, unless some early return will be taken in `canBeFinished` or `finishInternal` before it gets a chance to call `captureTransactionWithProfile` here: https://github.com/getsentry/sentry-cocoa/blob/0addc419e43e1ded284a76db8c7d2a64840b939e/Sources/Sentry/SentryTracer.m#L621
This would be a good test case.

Reply:
I verified that the profiler is still stopped in this case. Slightly before its own 30 second timeout. I think this will be simpler than trying to do it in a closure, which brings along the added complexity of weakself and cycles.